elasticdl_preprocessing/utils/analyzer_utils.py

```diff
@@ -0,0 +1,168 @@
+    max_value = os.getenv(env_name, None)
+    if max_value is None:
+        return default_value
+    else:
+        return float(max_value)
+
+
+def get_avg(feature_name, default_value):
+    """Get the average of numeric feature from the environment.
+    Return the default value if there is no the statistics in
+    the environment.
+
+    Args:
+        feature_name: String, feature name or column name in a table
+        default_value: Float.
+
+    Return:
+        Float
+    """
+    env_name = feature_name + "_avg"
+    mean = os.getenv(env_name, None)
+    if mean is None:
+        return default_value
+    else:
+        return float(mean)
+
+
+def get_stddev(feature_name, default_value):
+    """Get the standard deviation from the environment.
+    Return the default value if there is no the statistics in
+    the environment.
+
+    Args:
+        feature_name: String, feature name or column name in a table
+        default_value: Float.
+
+    Return:
+        Float.
+    """
+    env_name = feature_name + "_stddev"
+    std_dev = os.getenv(env_name, None)
+    if std_dev is None:
+        return default_value
+    else:
+        return float(std_dev)
+
+
+def get_bucket_boundaries(feature_name, default_value):
+    """Get the bucket boundaries from the environment.
+    Return the default value if there is no the statistics in
+    the environment.
+
+    Args:
+        feature_name: String, feature name or column name in a table
+        default_value: List with float values.
+
+    Return:
+        List with float values.
+    """
+    env_name = feature_name + "_bkt_boundaries"
+    boundaries = os.getenv(env_name, None)
+    if boundaries is None:
+        return default_value
+    else:
+        boundaries = list(map(float, boundaries.split(",")))
+        return sorted(set(boundaries))
+
+
+def get_distinct_count(feature_name, default_value):
+    """Get the count of distinct feature values set from the environment.
+    Return the default value if there is no the statistics in
+    the environment.
+
+    Args:
+        feature_name: String, feature name or column name in a table
+        default_value: Integer.
+
+    Return:
+        Integer.
+    """
+    env_name = feature_name + "_count"
+    count = os.getenv(env_name, None)
+    if count is None:
+        return default_value
+    else:
+        return int(count)
+
+
+def get_vocabulary(feature_name, default_value):
+    """Get the feature vocabulary from the environment.
+    Return the default value if there is no the statistics in
+    the environment.
+
+    Args:
+        feature_name: String, feature name or column name in a table
+        default_value: List with strings.
+
+    Return:
+        List with strings.
+    """
+    env_name = feature_name + "_vocab_path"
+    vocabulary_path = os.getenv(env_name, None)
+    if vocabulary_path is None:
+        return default_value
+    else:
+        return _read_vocabulary_from_file(vocabulary_path)
```

Comment:
How about returning file path directly?

Reply:
Returning a vocabulary list is more convenient for users to test the model. If return a file path, the users need to mock a file with vocabulary when they test the  model.

Reply:
User can pass either a vocabulary list or a file name.
I means that we don't convert it into a list if it's a file name, let the model or op to do this work.

Reply:
Good Suggestion!